File: webplayer/webplayer/resource/WebPacket.js
class WebPacket {
    constructor(packet) {
        if(packet == null){
            return;
        }
       this.streamIndex = packet.getStreamIndex();
       var packetData = packet.getData();
       packetData  = new Uint8Array(packetData);
       this.packetData = packetData;
       this.ptsMSec = packet.getPtsMSec();
       this.dtsMSec = packet.getDtsMSec();
       this.isKeyframe = packet.isKeyframe();
       this.isEnd = packet.isEnd();
       this.packetName = packet.packetName();
       this.mediaType = packet.getPacketType().value;
       this.packetPos = packet.getPacketPos();
       this.packetDuration = packet.getPacketDuration();
       this.packetFlag = packet.getPacketFlag();
       
    }

    convertToPacketFFmpeg(packetFFmpeg) {
        switch (this.mediaType) {
            case -1:
                this.mediaType = Module.PacketAVMediaType.AVTypeUnknow;
                break;
            case 0:
                this.mediaType = Module.PacketAVMediaType.AVTypeVideo;
                break;
            case 1:
                this.mediaType = Module.PacketAVMediaType.AVTypeAudio;
                break;
            case 2:
                this.mediaType = Module.PacketAVMediaType.AVTypeData;
                break;
            case 3:
                this.mediaType = Module.PacketAVMediaType.AVTypeSubtitle;
                break;
            case 4:
                this.mediaType = Module.PacketAVMediaType.AVTypeAttachment;
                break;
            default:
                this.mediaType = Module.PacketAVMediaType.AVTypeUnknow;
        }

        packetFFmpeg.setStreamIndex(this.streamIndex);
        packetFFmpeg.setData(this.packetData);
        packetFFmpeg.setPtsMSec(this.ptsMSec);
        packetFFmpeg.setDtsMSec(this.dtsMSec);
        packetFFmpeg.setKeyframeFlag(this.isKeyframe);

        packetFFmpeg.setEofFlag(this.isEnd);
        packetFFmpeg.setPacketType(this.mediaType);
        packetFFmpeg.setPacketPos(this.packetPos);
        packetFFmpeg.setPacketDuration(this.packetDuration);
        packetFFmpeg.setPacketFlag(this.packetFlag);

    }
}

File: webresource/webpacket.cpp
#include "webpacket.h"

using namespace emscripten;

EMSCRIPTEN_BINDINGS(WebPacketFFmpeg){
    class_<WebPacketFFmpeg>("WebPacketFFmpeg")
       .constructor<>()
       .function("AddRef", &WebPacketFFmpeg::AddRef)
       .function("Release", &WebPacketFFmpeg::Release)
       .function("setPtsMSec", &WebPacketFFmpeg::setPtsMSec)
       .function("getPtsMSec", &WebPacketFFmpeg::getPtsMSec)
       .function("setDtsMSec", &WebPacketFFmpeg::setDtsMSec)
       .function("getDtsMSec", &WebPacketFFmpeg::getDtsMSec)
       .function("setKeyframeFlag", &WebPacketFFmpeg::setKeyframeFlag)
       .function("isKeyframe", &WebPacketFFmpeg::isKeyframe)
       .function("setEofFlag", &WebPacketFFmpeg::setEofFlag)
       .function("isEnd", &WebPacketFFmpeg::isEnd)
       .function("packetName", &WebPacketFFmpeg::packetName)
       .function("getPacketFlag", &WebPacketFFmpeg::getPacketFlag)
       .function("setPacketFlag", &WebPacketFFmpeg::setPacketFlag)
       .function("getPacketPos", &WebPacketFFmpeg::getPacketPos)
       .function("setPacketPos", &WebPacketFFmpeg::setPacketPos)
       .function("getPacketDuration", &WebPacketFFmpeg::getPacketDuration)
       .function("setPacketDuration", &WebPacketFFmpeg::setPacketDuration)
       .function("setPacketType", &WebPacketFFmpeg::setPacketType)
       .function("getPacketType", &WebPacketFFmpeg::getPacketType)
       .function("getStreamIndex", &WebPacketFFmpeg::getStreamIndex)
       .function("getData", &WebPacketFFmpeg::getData)
       .function("setStreamIndex",&WebPacketFFmpeg::setStreamIndex)
       .function("setData",&WebPacketFFmpeg::setData)

       ;
}

EMSCRIPTEN_BINDINGS(PacketAVMediaType) {
    enum_<IPacket::AVMediaType>("PacketAVMediaType")
        .value("AVTypeUnknow", IPacket::AVTypeUnknow)
        .value("AVTypeVideo", IPacket::AVTypeVideo)
        .value("AVTypeAudio", IPacket::AVTypeAudio)
        .value("AVTypeData", IPacket::AVTypeData)
        .value("AVTypeSubtitle", IPacket::AVTypeSubtitle)
        .value("AVTypeAttachment", IPacket::AVTypeAttachment)
        ;
}